gubernator/github/classifier.py

```diff
@@ -206,12 +201,24 @@ def classify(events, statuses=None):
                 'xrefs': list of builds referenced (by GCS path),
             }
     '''
+    if events_iterator is not None:
+        events = next(events_iterator)
+
     merged = get_merged(events)
     labels = get_labels(events)
     comments = get_comments(events)
-    xrefs = get_xrefs(comments, merged)
-    approvers = get_approvers(comments)
     reviewers = get_reviewers(events)
+    distilled_events = distill_events(events)
+
+    if events_iterator is not None:
+        for events in events_iterator:
+            merged = get_merged(events, merged)
```

Comment:
Can't we do better than this? This seems awkwardly bolted on (aka we keep switching from one data structure to another)

Reply:
Alternative: have classify(events, status_fetcher=None) and classify_stream(events_iterator, status_fetcher), which both call a third classify_internal(merged,labels,comments,reviewers,distilled_events)?

Reply:
Sure SGTM